src/main/java/build/buildfarm/instance/shard/RedisShardBackplane.java

```diff
@@ -1419,19 +1432,245 @@ public boolean canPrequeue() throws IOException {
   }
 
   @Override
-  public BackplaneStatus backplaneStatus() throws IOException {
+  public BackplaneStatus backplaneStatus(Instance instance) throws IOException {
     return client.call(
         jedis ->
             BackplaneStatus.newBuilder()
                 .setPrequeue(prequeue.status(jedis))
                 .setOperationQueue(operationQueue.status(jedis))
                 .setBlockedActionsSize(blockedActions.size(jedis))
                 .setBlockedInvocationsSize(blockedInvocations.size(jedis))
-                .setDispatchedSize(jedis.hlen(config.getDispatchedOperationsHashName()))
+                .setDispatchedOperations(getDispatchedOperationsStatus(jedis, instance))
                 .addAllActiveWorkers(workerSet)
                 .build());
   }
 
+  /**
+   * @brief Analyze the currently dispatched operations to derive statistics about the operations
+   *     workers are currently executing.
+   * @details This status message is intended to be used as metrics.
+   * @param cluster An established redis cluster.
+   * @param instance An instance is used to get additional information about the operation.
+   * @return Metrics about the currently dispatched operations.
+   */
+  private DispatchedOperationsStatus getDispatchedOperationsStatus(
+      JedisCluster jedis, Instance instance) {
+    // Metrics related to dispatched operations
+    ActionAmounts actionAmounts = new ActionAmounts();
+    Integer requeuedOperationsAmount = 0;
+    Set<String> uniqueToolInvocationIds = Sets.newHashSet();
+    Map<String, Integer> fromQueueAmounts = new HashMap();
+    Map<String, Integer> toolAmounts = new HashMap();
+    Map<String, Integer> commandTools = new HashMap();
+    Map<String, Integer> actionMnemonics = new HashMap();
+    Map<String, Integer> targetIds = new HashMap();
+    Map<String, Integer> configIds = new HashMap();
+    Map<String, Integer> platformProperties = new HashMap();
```

Comment:
Do we need to keep track of these if we only care about the size for metrics? Why don't we just set the counters directly using labels that the prometheus library provides, just like we do for queue sizes (cpu vs gpu).

Reply:
You're suggesting we put Prometheus types into the backplane and populate them directly here (avoiding the maps)?  We could do that.  

Building up these maps is similar to what we already do for the cpu/gpu queue.  the backplane returns a  proto message, and the instance publishes it.  For queues (cpu/gpu), the backplane returns a `QueueStatus`.  For these metrics, the backplane returns a `DispatchedOperationsStatus`.

Reply:
Ok, I think this should work. Let's just keep it on the instance level.